torchvision/datasets/clevr.py

```diff
@@ -0,0 +1,88 @@
+import json
+import pathlib
+from typing import Any, Callable, Optional, Tuple, List
+from urllib.parse import urlparse
+
+from PIL import Image
+
+from .utils import download_and_extract_archive, verify_str_arg
+from .vision import VisionDataset
+
+
+class CLEVRClassification(VisionDataset):
+    """`CLEVR <https://cs.stanford.edu/people/jcjohns/clevr/>`_  classification dataset.
+
+    The number of objects in a scene are used as label.
+
+    Args:
+        root (string): Root directory of dataset where directory ``clevr`` exists or will be saved to if download is
+            set to True.
+        split (string, optional): The dataset split, supports ``"train"`` (default), ``"val"``, or ``"test"``.
+        transform (callable, optional): A function/transform that takes in an PIL image and returns a transformed
+            version. E.g, ``transforms.RandomCrop``
+        target_transform (callable, optional): A function/transform that takes in them target and transforms it.
+        download (bool, optional): If true, downloads the dataset from the internet and puts it in root directory. If
```

Comment:
> In other datasets like DTD, the `download` parameter comes before the transforms parameters. I feel like we should be consistent here.

This is inconsistent across datasets:

https://github.com/pytorch/vision/blob/5c9c8352ece2d5924017bc000f40e391eb06a482/torchvision/datasets/caltech.py#L39-L41

https://github.com/pytorch/vision/blob/5c9c8352ece2d5924017bc000f40e391eb06a482/torchvision/datasets/voc.py#L74-L77

From quickly looking through the source, it seems `download` last is more prevalent. So if we want to standardize here, I would go for that.

> I also wouldn't mind introducing the `*` for keyword-only parameters in these new datasets TBH.

The prototype datasets will no longer have a `download` flag, since the data is managed by `torchvision`. For everything else, we already have them as keyword only there:

https://github.com/pytorch/vision/blob/5c9c8352ece2d5924017bc000f40e391eb06a482/torchvision/prototype/datasets/_api.py#L60-L66

Reply:
> From quickly looking through the source, it seems download last is more prevalent. So if we want to standardize here, I would go for that

Ok, that sounds good. All other PRs that I've reviewed about the FLAVA datasets, `download` comes before though. So we'll need to update them before merging, and modify `Food101` as well as `DTD`.

> The prototype datasets will no longer have a download flag, since the data is managed by torchvision. For everything else, we already have them as keyword only there:

Sorry for being unclear, I was suggesting to use the `*` for these newly-introduced datasets, even if they're not in the `prototype` area.

Reply:
xref #5164